pkg/util/util.go

```diff
@@ -76,3 +76,13 @@ func CheckEnvBool(envBool string) bool {
 	}
 	return false
 }
+
+// IsM1 returns true if running on M1 architecture
+// returns false if not running on M1 architecture
+// We use this to setup longerHealthCheck
+func IsM1(myOS, myArch string) bool {
```

Comment:
based on this article https://www.yellowduck.be/posts/detecting-apple-silicon-via-go 
proper way to check it by calling:
```
sysctl -in sysctl.proc_translated
```
I just tested on my macs and it returns "" for Intel and != "" (`0`) for Intel chips.

Since we are using go.1.18 your approach should work I susspect:

> Go 1.17 [implemented](https://tip.golang.org/doc/go1.17#compiler) a new way of passing function arguments and 
results using registers instead of the stack on 64-bit x86 architecture on selected operating systems. Go 1.18 expands the supported platforms to include 64-bit ARM (GOARCH=arm64), big- and little-endian 64-bit PowerPC (GOARCH=ppc64, ppc64le), as well as 64-bit x86 architecture (GOARCH=amd64) on all operating systems. On 64-bit ARM and 64-bit PowerPC systems, benchmarking shows typical performance improvements of 10% or more.

https://tip.golang.org/doc/go1.18

Reply:
@andriisoldatenko it works provided that the user has installed the correct golang for their architecture.